tests/unit/test_chainer.py

```diff
@@ -393,3 +419,46 @@ def test_attach_wrong_framework(sagemaker_session):
         Chainer.attach(training_job_name='neo', sagemaker_session=sagemaker_session)
     assert "didn't use image for requested framework" in str(error)
+
+
+def test_attach_custom_image(sagemaker_session):
+    training_image = '1.dkr.ecr.us-west-2.amazonaws.com/my_custom_chainer_image:latest'
+    returned_job_description = {'AlgorithmSpecification':
+                                {'TrainingInputMode': 'File',
+                                 'TrainingImage': training_image},
+                                'HyperParameters':
+                                    {'sagemaker_submit_directory': '"s3://some/sourcedir.tar.gz"',
+                                     'sagemaker_program': '"iris-dnn-classifier.py"',
+                                     'sagemaker_s3_uri_training': '"sagemaker-3/integ-test-data/tf_iris"',
+                                     'sagemaker_enable_cloudwatch_metrics': 'false',
+                                     'sagemaker_container_log_level': '"logging.INFO"',
+                                     'sagemaker_job_name': '"neo"',
+                                     'training_steps': '100',
+                                     'sagemaker_region': '"us-west-2"'},
+                                'RoleArn': 'arn:aws:iam::366:role/SageMakerRole',
+                                'ResourceConfig':
+                                    {'VolumeSizeInGB': 30,
+                                     'InstanceCount': 1,
+                                     'InstanceType': 'ml.c4.xlarge'},
+                                'StoppingCondition': {'MaxRuntimeInSeconds': 24 * 60 * 60},
+                                'TrainingJobName': 'neo',
+                                'TrainingJobStatus': 'Completed',
+                                'OutputDataConfig': {'KmsKeyId': '',
+                                                     'S3OutputPath': 's3://place/output/neo'},
+                                'TrainingJobOutput': {'S3TrainingJobOutput': 's3://here/output.tar.gz'}}
+    sagemaker_session.sagemaker_client.describe_training_job = Mock(name='describe_training_job',
+                                                                    return_value=returned_job_description)
+
+    estimator = Chainer.attach(training_job_name='neo', sagemaker_session=sagemaker_session)
+    assert estimator.latest_training_job.job_name == 'neo'
+    assert estimator.role == 'arn:aws:iam::366:role/SageMakerRole'
+    assert estimator.train_instance_count == 1
+    assert estimator.train_max_run == 24 * 60 * 60
+    assert estimator.input_mode == 'File'
+    assert estimator.base_job_name == 'neo'
+    assert estimator.output_path == 's3://place/output/neo'
+    assert estimator.output_kms_key == ''
+    assert estimator.hyperparameters()['training_steps'] == '100'
+    assert estimator.source_dir == 's3://some/sourcedir.tar.gz'
+    assert estimator.entry_point == 'iris-dnn-classifier.py'
+    assert estimator.train_image() == training_image
```

Comment:
Same comment here about keeping minimal asserts (unless there's a specific reason that these interact with overriding the image.)